cf_xarray/accessor.py

```diff
@@ -586,6 +650,8 @@ def check_results(names, key):
         measures = []
         warnings.warn("Ignoring bad cell_measures attribute.", UserWarning)
 
+    custom_criteria = ChainMap(*OPTIONS["custom_criteria"])
```

Comment:
`ChainMap(*OPTIONS["custom_criteria"])` is always called both here and by `_get_custom_criteria`.
I'm wondering whether it makes sense to try to optimize it .
Maybe we could do something like what we are doing with `_all_cell_measures` and `_get_all_cell_measures`, where we define `_all_cell_measures = None` when we initialize the accessor, and we call `_get_all_cell_measures` only `if _all_cell_measures is None`)

Reply:
I think this is OK for now; I want to refactor this method eventually.